core/services/chainlink/application.go

```diff
@@ -258,6 +266,9 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 
 			srvcs = append(srvcs, wfLauncher, registrySyncer)
 		}
+	} else {
+		// If registry syncer is not set up we use a dummy local registry so that local capabilities can still be used
```

Comment:
@archseer this shouldn't be fallback behaviour: this should only be set explicitly if a specific config flag is set (eg. if the registryAddress == "__MOCK_REGISTRY_ADDRESS__" or similar

Reply:
Why not though? If no registry is set the node is in a broken state where local capabilities won't work. By providing a mock we're at least able to use workflows